prow/tide/status.go

```diff
@@ -83,23 +80,40 @@ type statusController struct {
 	// the minimum status update period.
 	lastSyncStart time.Time
 
-	// dontUpdateStatus contains all PRs for which the Tide sync controller
-	// updated the status to success prior to merging. As the name suggests,
-	// the status controller must not update their status.
-	dontUpdateStatus threadSafePRSet
-
 	sync.Mutex
 	poolPRs          map[string]CodeReviewCommon
 	requiredContexts map[string][]string
 	blocks           blockers.Blockers
 	baseSHAs         map[string]string
 
+	// Shared fields with sync controller
+	// dontUpdateStatus contains all PRs for which the Tide sync controller
+	// updated the status to success prior to merging. As the name suggests,
+	// the status controller must not update their status.
+	dontUpdateStatus *threadSafePRSet
+	// newPoolPending is a size 1 chan that signals that the main Tide loop has
+	// updated the 'poolPRs' field with a freshly updated pool.
+	newPoolPending chan bool
+	statusUpdate   *statusUpdate
+
 	storedState     map[string]storedState
 	storedStateLock sync.Mutex
 	opener          io.Opener
 	path            string
 }
 
+// statusUpdate contains the required fields from syncController when there is a
+// pending pool update.
+//
+// statusController will use the values from syncController blindly.
+type statusUpdate struct {
+	blocks           blockers.Blockers
+	poolPRs          map[string]CodeReviewCommon
+	baseSHAs         map[string]string
+	requiredContexts map[string][]string
+	sync.Mutex
+}
```

Comment:
This is redundant with the fields in `statusController`.

Reply:
inlined it